### ISSUE TITLE
[Question] Curriculum learning `env.common_step_counter`

### ISSUE THREAD

**Issue by elvisbreit** [isaac-sim/IsaacLab#3444]
### Question

I want to use Curriculum Learning for my task. By using the [curriculum utilities](https://isaac-sim.github.io/IsaacLab/main/source/how-to/curriculums.html#curriculum-utilities), I need to set the number of simulation steps after which a reward weight is changed. The variable `env.common_step_counter` is defined in the function, but I cannot find any description of it in the documentation. What exactly does _common_ mean in this context? 

Besides, if I use the rsl_rl library to train the agent and want reward changes to be applied after a specific number of iterations (policy updates), how can I determine the corresponding number of steps which I need to set?

**Comment by Mayankm96** [comment]
`common_step_counter` simply tracks the total number of `env.step(...)` calls across training. In other words, it counts how many environment steps have been executed.

If you’re using a PPO-based library like rsl_rl and want your curriculum change to happen after a certain number of policy updates (iterations), you can convert that into environment steps as follows:

```
change_after_step = agent_cfg.runner.num_steps_per_env * desired_iterations
```

**Comment by elvisbreit** [comment]
Hi @Mayankm96, thank you for helping out! I do have a few additional questions:

If I use more than one environment for training and `common_step_counter` counts all steps of `env.step(...)` for each environment, then when using the rsl_rl library to determine the step at which the curriculum should change, I would need to additionally multiply by `num_envs`, correct?

In the case of using the rl_games library, I would use `horizon_length` and `num_envs` together with the desired number of epochs to calculate the step after which the change should happen, right?

**Comment by Mayankm96** [comment]
The value `common_step_counter` represents the total number of step counts across all environments:
> https://github.com/isaac-sim/IsaacLab/blob/main/source/isaaclab/isaaclab/envs/manager_based_rl_env.py#L203

For RL-Games, the `horizon length` is equal to `num_steps_per_env` in RSL-RL. Essentially, this attribute specifies how many step calls are made to fill the online trajectory buffer before performing a policy update. Because of this, you don’t need to factor in the number of environments or epochs when interpreting it.

In pseudo-code, it looks like this:
```python
for n in range(num_interactions):
    for step_idx in range(num_steps_per_env):
        # step environment
        env.step(...)
        
        # record data
        agent.store(...)

    # perform learning
    for e in range(num_epochs):
       # backprop
       agent.update(...)
````

**Comment by elvisbreit** [comment]
Thank you for explaining @Mayankm96, it's clear for me now
